External/Plugins/ProjectManager/Controls/OpenResourceForm.cs

```diff
@@ -542,73 +574,85 @@ public static List<string> getMatchedItems(List<string> source, string searchTex
             return results;
         }
 
-        static bool AdvancedSearchMatch(string file, string searchText)
+        static double ScoreFileName(string str, string query, bool caseMatters)
         {
-            if (!string.Equals(searchText.ToUpperInvariant(), searchText)) return false;
-
-            int i = 0; int j = 0;
-            if (file.Length < searchText.Length) return false;
-            var text = file.ToCharArray();
-            var pattern = searchText.ToCharArray();
-            while (i < pattern.Length)
-            {
-                while (i < pattern.Length && j < text.Length && pattern[i] == text[j])
-                {
-                    i++;
-                    j++;
-                }
-                if (i == pattern.Length) return true;
-                if (char.IsLower(pattern[i])) return false;
-                while (j < text.Length && char.IsLower(text[j]))
-                {
-                    j++;
-                }
-                if (j == text.Length) return false;
-                if (pattern[i] != text[j]) return false;
-            }
-            return i == pattern.Length;
+            // Prefer shorter results. The effect is less pronounced when query length is increased.
+            return Score(str, query, caseMatters) / (str.Length + query.Length);
```

Comment:
Previously score was divided by an arithmetic average of `str.Length` and `query.Length`. Just using their `sum` is as good, though, since scores are only compared to each other, any constant factor (`1/2`) doesn't matter.